src/session/query.c

```diff
@@ -41,7 +50,10 @@ _Bool _z_pending_reply_eq(const _z_pending_reply_t *one, const _z_pending_reply_
 
 void _z_pending_reply_clear(_z_pending_reply_t *pr) {
     // Free reply
-    _z_reply_clear(&pr->_reply);
+    if (pr->_reply != NULL) {
+        _z_reply_clear(pr->_reply);
+        z_free(pr->_reply);
```

Comment:
`_z_reply_free` is the same as these two operations.
Use it instead.

This applies to any type that you can clear you can also free it.